src/istio/chart/values.yaml

```diff
@@ -4,19 +4,25 @@ name: change-me
 # Domain name for the gateway
 domain: "###ZARF_VAR_DOMAIN###"
 
-# Hosts to add to this gateway
-hosts:
-  - "*"
+# tls:
+#   # Name of the gateway port to use for TLS
+#   - name: https
```

Comment:
Changing this to a list of TLS makes a ton of sense but it does make the overrides for bundles a bit gross I think (targeting specific list items, merging things, etc). I think we can easily switch this to a map where the key is the name, as long as we adjust the templates accordingly.
```yaml
tls:
  https:
    # The TLS mode for the gateway. One of `SIMPLE`, `MUTUAL`, 'OPTIONAL_MUTUAL', `PASSTHROUGH`
    mode: SIMPLE

    # Hosts to add to this gateway
    hosts:
      - "*"

    # The secret name for the TLS certificate when not using passthrough mode
    secretName: gw-cert

    # For TLS SIMPLE mode, the TLS certificate data for the gateway (base64 encoded)
    cert: ""

    # For TLS SIMPLE mode, the TLS key data for the gateway (base64 encoded)
    key: ""

    # For mTLS mode, the CA certificate data for the gateway (base64 encoded)
    cacert: ""
```
For your example one it would look like this, additional ones for keycloak, etc would just be `tls.keycloak`. On the template side `range` will work the same, we'll just need `{{ range $name,$tls := .Values.tls }}` to capture the key (`$name`), and then update any references to `$tls.name` to `$name`.

Should be almost entirely equivalent, but would make it a bit easier on the bundle side to target `tls.keycloak.cacert` for example.